src/cache_padded.rs

```diff
@@ -58,63 +42,102 @@ zeros_valid!(i8 i16 i32 i64 isize);
-    pub fn zeroed() -> CachePadded<T> {
-        CachePadded {
-            data: UnsafeCell::new(([0; CACHE_LINE])),
-            _marker: ([], marker::PhantomData),
-        }
-    }
+/// Assert that the size and alignment of `T` are consistent with `CachePadded<T>`.
+fn assert_valid<T>() {
+    assert!(mem::size_of::<T>() <= mem::size_of::<CachePadded<T>>());
+    assert!(mem::align_of::<T>() <= mem::align_of::<CachePadded<T>>());
+}
+
+/// Pad `T` to the length of a cacheline.
+///
+/// Sometimes concurrent programming requires a piece of data to be padded out to the size of a
+/// cacheline to avoid "false sharing": cachelines being invalidated due to unrelated concurrent
+/// activity. Use the `CachePadded` type when you want to *avoid* cache locality.
+///
+/// # Warning
+///
+/// - The wrapped data is never dropped; move out using `ptr::read` if you need to run dtors.
+/// - Do not rely on this actually being padded to the cache line for the correctness of your
+///   program. Certain compiler options or hardware might actually cause it to not be aligned.
+pub struct CachePadded<T> {
+    /// The data.
+    ///
+    /// This is untyped on purpose in order to force a certain memory representation.
+    data: UnsafeCell<[u8; CACHE_LINE]>,
+    /// The memory padding.
+    _pad: Pad,
+    /// The typed marker.
+    _marker: marker::PhantomData<T>,
+}
 
-    /// A const fn equivalent to mem::zeroed().
+impl<T: ZerosValid> CachePadded<T> {
+    /// Create a zeroed cache padded value.
     #[cfg(feature = "nightly")]
     pub const fn zeroed() -> CachePadded<T> {
         CachePadded {
-            data: UnsafeCell::new(([0; CACHE_LINE])),
-            _marker: ([], marker::PhantomData),
+            data: UnsafeCell::new([0; CACHE_LINE]),
+            _pad: [],
+            _marker: marker::PhantomData,
         }
     }
 }
 
-#[inline]
-/// Assert that the size and alignment of `T` are consistent with `CachePadded<T>`.
-fn assert_valid<T>() {
-    assert!(mem::size_of::<T>() <= mem::size_of::<CachePadded<T>>());
-    assert!(mem::align_of::<T>() <= mem::align_of::<CachePadded<T>>());
+impl<T: ZerosValid> Default for CachePadded<T> {
+    fn default() -> CachePadded<T> {
+        CachePadded {
+            data: UnsafeCell::new([0; CACHE_LINE]),
+            _pad: [],
+            _marker: marker::PhantomData,
+        }
+    }
 }
 
 impl<T> CachePadded<T> {
-    /// Wrap `t` with cacheline padding.
+    /// Wrap a value in a type padding it to a cache line.
+    ///
+    /// # Warning
+    ///
+    /// The wrapped data is never dropped; move out using `ptr:read` if you need to run dtors.
     ///
-    /// **Warning**: the wrapped data is never dropped; move out using
-    /// `ptr:read` if you need to run dtors.
+    /// # Panics
+    ///
+    /// If `T` is incompatible with `CachePadded<T>` (e.g. too big), this will hit an assertion.
     pub fn new(t: T) -> CachePadded<T> {
+        // Assert the validity.
```

Comment:
@konstin "it's somebodies coding style" is not an argument. It directly impedes readability to labal things redundantly. Any piece of the source code, even if it is a comment, should pay its weight or go to the chopping block. Most software source code metrics such as cyclomatic complexity and number of bugs directly correlate with the numbers of line of code. More code means more room for bugs and out of date documentation.